### PR TITLE
now support validating bodies to ensure all references exist in inline array

### PR DESCRIPTION
**Pull request by thekemkid**
Card: https://app.clubhouse.io/condenastinternational/story/11551/flyway-service-should-error-when-updating-inline-entity-hrefs-if-entity-is-not-found-in-article-inline-array

### CODE REVIEW

File: models/article.js [hunk 1]
```diff
@@ -8,11 +8,63 @@ const categoryModel = require('./category');
 const validContentSources = require('./contentSources');
 const pathType = require('./path_type');
 
+const customJoi = joi.extend(j => ({
```
Comment (andrewcashmore) [inline models/article.js]: Name this a little more meaningfully

File: models/article.js [hunk 2]
```diff
@@ -8,11 +8,63 @@ const categoryModel = require('./category');
...
+  language: {
+    embeddedEntities:
+      'can only contain references to entities existing in inline array [bad reference within string: "{{href}}"]',
+    embeddedEntitiesNesting: 'can only be nested within an object'
+  },
+  rules: [
+    {
+      name: 'embeddedEntities',
+      validate(params, value, state, options) {
+        // console.log(this, arguments);
```
Comment (andrewcashmore) [inline models/article.js]: Remove this?

**Comment by andrewcashmore** [comment]
This looks alright to me, I know the custom rule is quite verbose to create but it seems like the right way to do it. Just a couple of little comments

File: models/article.js [hunk 3]
```diff
@@ -8,11 +8,63 @@ const categoryModel = require('./category');
...
+            `string.embeddedEntitiesNesting`,
+            {
+              v: value
+            },
+            state,
+            options
+          );
+        }
+        const inline = article.inline || [];
+        const inlineHrefs = new Set(inline.map(item => item.inlineHref));
```
Comment (almostagile) [inline models/article.js]: `inlineHref` is optional. You need to generate the default for the item if it is not supplied (e.g. `/photos/${id}` for inline photo).

File: models/article.js [hunk 4]
```diff
@@ -8,11 +8,62 @@ const categoryModel = require('./category');
...
+  base: j.string(),
+  name: 'string',
+  language: {
+    embeddedEntities:
+      'can only contain references to entities existing in inline array [bad reference within string: "{{href}}"]',
+    embeddedEntitiesNesting: 'can only be nested within an object'
+  },
+  rules: [
+    {
+      name: 'embeddedEntities',
```
Comment (almostagile) [inline models/article.js]: Maybe `allowInlineEmbeds` might read better

File: models/article.js [hunk 5]
```diff
@@ -8,11 +8,62 @@ const categoryModel = require('./category');
 const validContentSources = require('./contentSources');
 const pathType = require('./path_type');
 
+const extendedStringValidator = joi.extend(j => ({
+  base: j.string(),
+  name: 'string',
```
Comment (almostagile) [inline models/article.js]: Maybe name this `body`

**Pull request merged**